ethexe/service/src/lib.rs

```diff
@@ -154,12 +169,36 @@ impl Service {
             .with_context(|| "failed to query validators")?;
         log::info!("👥 Validators set: {validators:?}");
 
+        let maybe_validator_identifiers: Result<Vec<_>, _> = validators
+            .iter()
+            .map(|address| Identifier::deserialize(&ActorId::from(*address).into_bytes()))
+            .collect();
+        let validator_identifiers = maybe_validator_identifiers.expect("conversion failed");
+        let identifiers = validator_identifiers.into_iter().collect();
+
+        let validators_verifiable_secret_sharing_commitment = router_query
+            .validators_verifiable_secret_sharing_commitment()
+            .await
+            .with_context(|| "failed to query validators verifiable secret sharing commitment")?;
+        let verifiable_secret_sharing_commitment =
+            VerifiableSecretSharingCommitment::deserialize_whole(
+                &validators_verifiable_secret_sharing_commitment,
+            )
+            .with_context(|| "failed to decode VerifiableSecretSharingCommitment")?;
+
+        let public_key_package =
```

Comment:
also makes sense to check that `public_key_package.verifying_key() == Router.validatorsAggregatedPublicKey()`. if condition is not met, then we don't even run the sequencer/validator and consider that whole network has been compromised